x-pack/plugins/event_log/server/es/cluster_client_adapter.ts

```diff
@@ -30,14 +42,71 @@ export interface QueryEventsBySavedObjectResult {
 export class ClusterClientAdapter {
   private readonly logger: Logger;
   private readonly clusterClientPromise: Promise<EsClusterClient>;
+  private readonly docBuffer$: Subject<Doc>;
+  private readonly docsBufferedSubscription: Subscription;
+  private readonly doneWriting: ReadySignal;
+  private readonly context: EsContext;
 
   constructor(opts: ConstructorOpts) {
     this.logger = opts.logger;
     this.clusterClientPromise = opts.clusterClientPromise;
+    this.context = opts.context;
+    this.docBuffer$ = new Subject<Doc>();
+    this.doneWriting = createReadySignal();
+
+    // buffer event log docs for time / buffer length, ignore empty
+    // buffers, then index the buffered docs
+    const docsBuffered$ = this.docBuffer$.pipe(
+      bufferTime(EVENT_BUFFER_TIME, null, EVENT_BUFFER_LENGTH),
+      filter((docs) => docs.length > 0),
+      switchMap(async (docs) => await this.indexDocuments(docs))
+    );
```

Comment:
You can actually use this instead of the `ReadySignal`.

All Observables expose a `toPromise` which returns a promise which resolves when they complete.
As `docBuffered$` completes when the `docBuffer$` completes, you can do something like:

```
  this.docBufferedFlushed = docsBuffered$.toPromise();
```

and then in `shutdown()` you can do:

```
wait  this.docBufferedFlushed;
```